tests/functional/spec/list_collector_section_summary_items.spec.js

```diff
@@ -0,0 +1,190 @@
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesNo();
+      answerUkBasedQuestion();
+      $(SectionSummaryPage.companiesListRemoveLink(1)).click();
+      expect(browser.getUrl()).to.contain("remove-company/?return_to=section-summary");
+    });
+    it("When I add my own item and make it not on the path, Then the list of answers should not be visible on the section summary.", () => {
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesNo();
+      answerUkBasedQuestion();
+      removeFirstCompany();
+      expect(browser.getUrl()).to.contain(SectionSummaryPage.url());
+      expect($(SectionSummaryPage.companiesListEditLink(1)).isExisting()).to.be.false;
+      expect($(SectionSummaryPage.companiesListRemoveLink(1)).isExisting()).to.be.false;
+    });
+    it("When I remove my own item but list collector is still on the path, Then the placeholder text should be visible on the section summary.", () => {
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesNo();
+      answerUkBasedQuestion();
+      removeFirstCompany();
+      expect(browser.getUrl()).to.contain(SectionSummaryPage.url());
+      expect($("body").getText()).to.contain("No UK company or branch added");
+    });
+    it("When I add my own item and relevant data, Then after I answer no on additional items page I should get to the section summary page.", () => {
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesNo();
+      answerUkBasedQuestion();
+      expect(browser.getUrl()).to.contain(SectionSummaryPage.url());
+    });
+    it("When I add my own item and relevant data, Then after I answer yes on additional items page I should be able to choose an item from the items list and add relevant data about it.", () => {
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesYes();
+      expect($(AnyCompaniesOrBranchesAddPage.companyOrBranchName()).isExisting()).to.be.true;
+      expect($(AnyCompaniesOrBranchesAddPage.registrationNumber()).isExisting()).to.be.true;
+      expect($(AnyCompaniesOrBranchesAddPage.authorisedInsurerRadioYes()).isExisting()).to.be.true;
+      expect($(AnyCompaniesOrBranchesAddPage.heading()).getText()).to.contain(
+        "Give details about the company or branch that undertakes general insurance business"
+      );
+    });
+    it("When I add my own item and relevant data, Then I should be able to edit that item from the section summary page.", () => {
+      drivingQuestionYes();
+      addCompany("Company A", "123", true);
+      anyMoreCompaniesNo();
+      answerUkBasedQuestion();
+      $(SectionSummaryPage.companiesListEditLink(1)).click();
+      expect(browser.getUrl()).to.contain("edit-company/?return_to=section-summary");
```

Comment:
What do we do for existing test?. This tests you go to the edit page, but not necessarily for the right item. Is that in scope?. How do we test currently?

Reply:
Updated the test now 👍